eng/common/docgeneration/templates/matthews/styles/main.js

```diff
@@ -0,0 +1,224 @@
+    httpGetAsync(targetUrl, function (responseText) {
+        if (responseText) {
+            version = responseText.match(/[^\r\n]+/g)
+            $(latestVersions).append('<li><a href="' + getPackageUrl(SELECTED_LANGUAGE, packageName, version) + '" target="_blank">' + version + '</a></li>')
+        }             
+    })
+}
+
+function populateIndexList(selector, packageName) {
+    var url = "https://azuresdkdocs.blob.core.windows.net/$web/" + SELECTED_LANGUAGE + "/" + packageName + "/versioning/versions"
+    var latestGAUrl = "https://azuresdkdocs.blob.core.windows.net/$web/" + SELECTED_LANGUAGE + "/" + packageName + "/versioning/latest-ga"
+    var latestPreviewUrl = "https://azuresdkdocs.blob.core.windows.net/$web/" + SELECTED_LANGUAGE + "/" + packageName + "/versioning/latest-preview"
+    var latestVersions = document.createElement("ul")
+    httpGetLatestAsync(latestGAUrl, latestVersions, packageName)
+    httpGetLatestAsync(latestPreviewUrl, latestVersions, packageName)
+    var publishedVersions = $('<ul style="display: none;"></ul>')
+    var collapsible = $('<div class="versionarrow">&nbsp;&nbsp;&nbsp;Other versions</div>')
+
+    $(selector).next().after(latestVersions)
+    $(latestVersions).after(collapsible)
+    $(collapsible).after(publishedVersions)
+    // Add collapsible arrows on versioned docs.
+    $(collapsible).on('click', function(event) {
+        event.preventDefault();
+        if (collapsible.hasClass('disable')) {
+            return
+        }
+        $(this).toggleClass('down')
+        if ($(this).hasClass('down')) {
+            if (!$(selector).hasClass('loaded')){
+                httpGetAsync(url, function (responseText) {
+                    if (responseText) {
+                        options = responseText.match(/[^\r\n]+/g)
+                        for (var i in options) {
+                            $(publishedVersions).append('<li><a href="' + getPackageUrl(SELECTED_LANGUAGE, packageName, options[i]) + '" target="_blank">' + options[i] + '</a></li>')
+    
+                        }
+                    }
+                    else {
+                        $(publishedVersions).append('<li>No discovered versions present in blob storage.</li>')
+                    }                
+                    $(selector).addClass("loaded")
+                })
+            }
+            $(publishedVersions).show()
+        } else {
+            $(publishedVersions).hide()
+        }
+    });
+}
+
+function getPackageUrl(language, package, version) {
+    return "https://azuresdkdocs.blob.core.windows.net/$web/" + language + "/" + package + "/" + version + "/" + INDEX_HTML
+}
+
+// Populate Versions
+$(function () {
+    if (WINDOW_CONTENTS.length < 7 && WINDOW_CONTENTS[WINDOW_CONTENTS.length - 1] != 'index.html') {
+        console.log("Run PopulateList")
+
+        $('h4').each(function () {
+            var pkgName = $(this).text().replace(PACKAGE_REGEX, PACKAGE_REPLACEMENT)
```

Comment:
Does this work correctly if the REGEX is empty?

Also I assume you added this to handle the JS case correct?